saleor/graphql/account/mutations/authentication/create_token.py

```diff
@@ -108,15 +109,20 @@ def perform_mutation(  # type: ignore[override]
             additional_paylod["aud"] = f"custom:{audience}"
             refresh_additional_payload["aud"] = f"custom:{audience}"
 
+        user = cls.get_user(info, email, password)
         access_token = create_access_token(user, additional_payload=additional_paylod)
         refresh_token = create_refresh_token(
             user, additional_payload=refresh_additional_payload
         )
         setattr(info.context, "refresh_token", refresh_token)
         info.context.user = user
         info.context._cached_user = user
-        user.last_login = timezone.now()
-        user.save(update_fields=["last_login", "updated_at"])
+        time_now = timezone.now()
+        threshold_delta = timedelta(seconds=settings.TOKEN_UPDATE_LAST_LOGIN_THRESHOLD)
+
+        if not user.last_login or user.last_login + threshold_delta < time_now:
+            user.last_login = time_now
+            user.save(update_fields=["last_login", "updated_at"])
```

Comment:
Maybe we should create a function from that? I see this is a port of changes, so maybe we can change it only on main